control/tap-ctl-spawn.c

```diff
@@ -172,8 +174,42 @@ tap_ctl_get_child_id(int readfd)
 	return id;
 }
 
+/* Move the process to the nominated cgroup slice */
+static int
+tap_ctl_move_to_cgroup(int pid, const char *slice)
+{
+	FILE *f = NULL;
+	char *path;
+
+	if (!slice)
+		return -EINVAL;
+
+	path = malloc(FILENAME_MAX);
+	if (!path) {
+		EPRINTF("malloc failed: %d\n", errno);
+		return -errno;
+	}
+
+	snprintf(path, FILENAME_MAX, "/sys/fs/cgroup/cpu/%s/tasks", slice);
+	path[FILENAME_MAX-1] = '\0';
```

Comment:
It's one of those things that all sensible implementations will enforce, but could cause headaches if this code was ever to be ported to other platforms. I always feel it's better to be defensive, but I'll happily remove it if you'd prefer.